mk/spksrc.service.installer.functions

```diff
@@ -75,7 +75,10 @@ install_python_virtualenv ()
     python3 -m venv --system-site-packages ${SYNOPKG_PKGDEST}/env
 
     # Update to latest pip package installer
-    python3 -m pip install --upgrade pip
+    if ! python3 -m pip install --upgrade pip; then
+    echo "Error: Python upgrade failed" 1>&2
+        exit 1
+    fi
```

Comment:
we should only log, but not exit on error.
the following set_unix_permissions is mandatory for DSM 6.

Reply:
Suggesting we always refer back to python packaged pip version... such as:
```
read -r -a PIP_VERSION 2>/dev/null <<< $(python3 -m pip --version)
[ "${PIP_VERSION[1]}" ] && PIP="pip==${PIP_VERSION[1]}" || PIP="pip"
if ! python3 -m pip install --upgrade ${PIP}; then
   echo "Error: Python upgrade failed" 1>&2
fi
```